src/storage/txn/scheduler.rs

```diff
@@ -431,15 +431,45 @@ impl<E: Engine, L: LockManager> Scheduler<E, L> {
         SCHED_STAGE_COUNTER_VEC.get(tag).read_finish.inc();
 
         debug!("read command finished"; "cid" => cid);
-        let tctx = self.inner.dequeue_task_context(cid);
-        if let ProcessResult::NextCommand { cmd } = pr {
+        if let ProcessResult::NextCommands { mut cmds } = pr {
```

Comment:
The logic here is very complex.
I wonder how can we improve these.

Reply:
I think @sticnarf is working on using async in the scheduler, etc.